OptimizelySDK/OptlyConfig/OptimizelyConfigService.cs

```diff
@@ -0,0 +1,238 @@
+                            experiment.Id,
+                            variation.FeatureEnabled ?? false,
+                            variation.FeatureVariableUsageInstances);
+
+                        var optimizelyVariation = new OptimizelyVariation(variation.Id,
+                            variation.Key,
+                            variation.FeatureEnabled,
+                            variablesMap);
+
+                        variationsMap.Add(variation.Key, optimizelyVariation);
+                    }
+                    var optimizelyExperiment = new OptimizelyExperiment(experiment.Id,
+                        experiment.Key,
+                        variationsMap);
+
+                    experimentsMap.Add(experiment.Key, optimizelyExperiment);
+                }
+            }
+
+            return experimentsMap;
+        }
+
+        /// <summary>
+        /// Make map of featureVariable which are associated with given feature experiment 
+        /// </summary>
+        /// <param name="projectConfig">The project config</param>
+        /// <param name="variableIdMap">Map containing variable ID as key and Object of featureVariable</param>
+        /// <param name="experimentId">experimentId of featureExperiment</param>
+        /// <param name="featureEnabled">featureEnabled of variation</param>
+        /// <param name="featureVariableUsageInstances">list of FeatureVariableUsage containing key and value</param>
+        /// <returns>Dictionary | Dictionary of FeatureVariable key and value as FeatureVariable object</returns>
+        private IDictionary<string, OptimizelyVariable> MergeFeatureVariables(
+           ProjectConfig projectConfig,
+           IDictionary<string, FeatureVariable> variableIdMap,
+           string experimentId,
+           bool featureEnabled,
+           List<FeatureVariableUsage> featureVariableUsageInstances)
+        {
+            var variablesMap = new Dictionary<string, OptimizelyVariable>();
+            var featureList = projectConfig.GetExperimentFeatureList(experimentId);
+            var featureIdVariablesMap = GetFeatureIdVariablesMap(projectConfig);
+            if (featureList != null)
+            {
+                if (featureVariableUsageInstances != null)
+                {
+                    featureVariableUsageInstances.ForEach(featureVariableUsage =>
+                    {
+                        var optimizelyVariable = new OptimizelyVariable(featureVariableUsage.Id,
+                            variableIdMap[featureVariableUsage.Id].Key,
+                            variableIdMap[featureVariableUsage.Id].Type.ToString().ToLower(),
+                            featureEnabled ? featureVariableUsage.Value : variableIdMap[featureVariableUsage.Id].DefaultValue);
+
+                        variablesMap.Add(variableIdMap[featureVariableUsage.Id].Key, optimizelyVariable);
+                    });
+                }
+
+                featureList.ForEach(featureId =>
```

Comment:
No need for loop here.
featureList should have at most one item - an experimentId is mapped to one feature or none.